pkg/operator/apis/monitoring/v1/http_types.go

```diff
@@ -20,40 +20,154 @@ import (
+		}
+		if ns != monitoringNamespace {
+			return "", fmt.Errorf("PodMonitoring secret selector can't select secret from the different namespace than PodMonitoring namespace %q, got: %q; Consider using ClusterPodMonitoring or copy secret to PodMonitoring namespace", m.GetNamespace(), ns)
+		}
+	}
+
+	ref := fmt.Sprintf("%s/%s/%s", ns, s.Name, s.Key)
+	pool.Set(ref, secrets.KubernetesSecretConfig{
+		Namespace: ns,
+		Name:      s.Name,
+		Key:       s.Key,
+	})
+	return ref, nil
+}
+
+// Auth sets the `Authorization` header on every HTTP request.
 type Auth struct {
-	// The authentication type. Defaults to Bearer, Basic will cause an error.
+	// The authentication type. Defaults to Bearer.
+	// Basic will cause an error, as the BasicAuth object should be used instead.
 	Type string `json:"type,omitempty"`
-	// TODO: Add credentials: https://github.com/GoogleCloudPlatform/prometheus-engine/issues/450
+	// Credentials references the Kubernetes secret's key with the credentials
+	// (token) for the auth header to send along the request.
+	// Optional, as in previous resource versions we allowed no credentials.
+	// +optional
+	Credentials *SecretSelector `json:"credentials,omitempty"`
 }
 
-func (c *Auth) ToPrometheusConfig() *config.Authorization {
-	return &config.Authorization{
-		Type: c.Type,
+func (c *Auth) ToPrometheusConfig(m PodMonitoringCRD, pool PrometheusSecretConfigs) (*config.Authorization, error) {
+	ref, err := c.Credentials.toPrometheusSecretRef(m, pool)
+	if err != nil {
+		return nil, err
 	}
+	return &config.Authorization{
+		Type:           c.Type,
+		CredentialsRef: ref,
+	}, nil
 }
 
-// BasicAuth sets the `Authorization` header on every scrape request with the
-// configured username.
-//
-// Currently the password is not configurable and always empty.
+// BasicAuth sets the `Authorization` header on every HTTP request with the
+// configured username and optional password.
 type BasicAuth struct {
 	// The username for authentication.
 	Username string `json:"username,omitempty"`
-	// TODO: Add password: https://github.com/GoogleCloudPlatform/prometheus-engine/issues/450
+	// Password references the Kubernetes secret's key with the password to use.
+	// Optional, as in previous resource versions we allowed no credentials.
+	// +optional
+	Password *SecretSelector `json:"password,omitempty"`
 }
 
-func (c *BasicAuth) ToPrometheusConfig() *config.BasicAuth {
-	return &config.BasicAuth{
-		Username: c.Username,
+func (c *BasicAuth) ToPrometheusConfig(m PodMonitoringCRD, pool PrometheusSecretConfigs) (*config.BasicAuth, error) {
+	ref, err := c.Password.toPrometheusSecretRef(m, pool)
+	if err != nil {
+		return nil, err
 	}
+	return &config.BasicAuth{
+		Username:    c.Username,
+		PasswordRef: ref,
+	}, nil
 }
 
-// TLS specifies TLS configuration parameters from Kubernetes resources.
+// TLS specifies TLS configuration used for HTTP requests.
 type TLS struct {
 	// Used to verify the hostname for the targets.
```

Comment:
```suggestion
	// ServerName is used to verify the hostname for the targets.
```

Reply:
BTW shouldn't all of those be optional? 🤔  I don't remember now but why sometimes we put explicit +optional sometimes not (for optional fields). It feels the behaviour is that if message has any +optional, other fields are marked as required? 🤔 

<img width="862" alt="image" src="https://github.com/GoogleCloudPlatform/prometheus-engine/assets/6950331/0ef0c64c-9a25-493c-aea5-8156fc90bf35">

Reply:
According to https://book.kubebuilder.io/reference/markers/crd-validation this only applies when using `+kubebuilder:validation:Required`. Since these are not the top-level CRD, I will include this annotation on all fields.